### PR TITLE
Email activity API Documentation

### LINKED ISSUE #562: Updates for the new Email Activity API (beta)
SendGrid has added an API to download information from the Email Activity feed. There are four new endpoints:
* [GET /messages](https://sendgrid.api-docs.io/v3.0/email-activity/filter-all-messages)
* [GET /messages/{msg_id}](https://sendgrid.api-docs.io/v3.0/email-activity/filter-messages-by-message-id)
* [POST /messages/download/](https://sendgrid.api-docs.io/v3.0/email-activity/request-csv)
* [GET /messages/download/{download_uuid}](https://sendgrid.api-docs.io/v3.0/email-activity/download-csv)

A getting started doc for the API is [here](https://sendgrid.com/docs/API_Reference/Web_API_v3/Tutorials/getting_started_email_activity_api.html).

The following items need to be updated in this repo:
* [USAGE.md](https://github.com/sendgrid/sendgrid-python/blob/master/USAGE.md)
* [/examples](https://github.com/sendgrid/sendgrid-python/tree/master/examples) (create a `messages` folder with a file `messages.py`)
* [Tests](https://github.com/sendgrid/sendgrid-python/blob/master/test/test_sendgrid.py)
* [USE_CASES.md](https://github.com/sendgrid/sendgrid-python/blob/master/USE_CASES.md)

### PR DESCRIPTION
**Pull request by dhoeric**
# Fixes #562 

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Add examples for Email Activity API
- Add usage for Email Activity API

If you have questions, please send an email to [SendGrid](mailto:dx@sendgrid.com), or file a GitHub Issue in this repository.

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=664) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=664) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-python?pullRequest=664) it.</sub>

**Comment by thinkingserious** [comment]
Hello @dhoeric,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by thinkingserious** [comment]
In addition to the updates requested by @misterdorm, 

Could you please merge this against our [v4 branch](https://github.com/sendgrid/sendgrid-python/tree/v4)? Thank you!

With Best Regards,

Elmer

**Comment by thinkingserious** [comment]
Hello @dhoeric,<br /><br/>Thanks again for the PR!<br /><br />                                         You have earned 6 out of the 5 points needed to receive glorious SendGrid Hacktoberfest swag.<br /><br />                                         Please take a moment to checkout <a href="https://dx.sendgrid.com/hacktoberfest/issues">this link</a> to find more issues to get you past the required threshold or to simply continue the celebration.<br /><br />                                         Also, <strong>please be sure you have officially registered with us <a href="https://dx.sendgrid.com/hacktoberfest/join">here</a> by November 1, 2018</strong> to qualify.<br /><br />                                         If you have any questions you can email us at <a href="mailto:dx+hacktoberfest2018@sendgrid.com">dx+hacktoberfest2018@sendgrid.com</a>.<br /><br />                                         Thank you and Happy Hacktobering!<br /><br />                                         Team SendGrid

File: USAGE.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ sg = SendGridAPIClient(os.environ.get('SENDGRID_API_KEY'))
 * [CLIENTS](#clients)
 * [CONTACTDB](#contactdb)
 * [DEVICES](#devices)
+* [EMAIL ACTIVITY](#email-activity)
```
Comment (thinkingserious) [inline USAGE.md:27]: I think we should move this to the [use cases](https://github.com/sendgrid/sendgrid-python/tree/main/use_cases) directory. This file is auto-generated from our [OpenAPI specification](https://github.com/sendgrid/sendgrid-oai).

**Review by thinkingserious — CHANGES_REQUESTED** [review]
Thank you for the PR once again! My apologies for the delayed code review.

**Comment by childish-sambino** [comment]
Closing due to inactivity. Please ping here if work is to continue.

**Pull request closed without merging**